docs/DeveloperGuide.md

```diff
@@ -308,32 +308,198 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
 ### Use cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is the `InSUREance` and the **Actor** is the `user`, unless specified otherwise)
 
-**Use case: Delete a person**
+---
+**Use case 01: Add a client**
 
 **MSS**
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to add a new client
+2. System checks new client details.
+3. System assigns client an ID.
+4. System shows successfully added a new client message.
 
-    Use case ends.
+Use case ends.
+
+**Extensions**
+
+* 2a. Client details are invalid.
+    * 2a1. System shows invalid client details error message.
+    * 2a2. User enters new client details.
+
+    Steps 2a1-2a2 are repeated until the data entered is correct
+
+    Use case resumes from step 3.
+
+* 2b. Client name is identical to another client that already exists inside the system.
+    * 2b1. System sends a warning about identical client to user.
+
+    Use case resumes from step 3.
+---
+<a name = "UC02"></a> **Use case 02: Select a client**
+
+**MSS**
+
+1. User requests to select a client.
+2. System checks if the index is valid.
+3. System selects the client.
+
+Use case ends.
+
+**Extensions**
+
+* 2a. Client index is invalid.
+    * 2a1. System shows invalid index invalid error message.
+    * 2a2. User enters new index.
+
+    Steps 2a1-2a2 are repeated until the index is valid.
+  
+    Use case resumes from step 3.
+---
+**Use case 03: Delete a client**
+
+**MSS**
+
+1.  User requests to list clients.
+2.  System shows a list of clients.
+3.  User [selects a client](#UC02) to delete.
+4.  System deletes the person.
+
+Use case ends.
 
 **Extensions**
 
 * 2a. The list is empty.
 
-  Use case ends.
+Use case ends.
 
-* 3a. The given index is invalid.
+---
+<a name = "UC04"></a>**Use case 04: Select an insurance plan**
 
-    * 3a1. AddressBook shows an error message.
+**MSS**
 
-      Use case resumes at step 2.
+1.  User selects an insurance plan ID.
+2.  System checks if the ID is valid.
+3.  System has selected the insurance plan ID.
 
-*{More to be added}*
+Use case ends.
+
+**Extensions**
+
+* 3a. The client does not have the specified insurance plan.
+    * 3a1. System shows an error message to user.
```

Comment:
May I know what is this extension here for? I think you may have considered this extension under UC06 instead